modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/messaging/BatchEngineTaskListener.java

```diff
@@ -0,0 +1,159 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.batch.engine.internal.messaging;
+
+import com.liferay.batch.engine.BatchEngineTaskExecuteStatus;
+import com.liferay.batch.engine.BatchEngineTaskExecutor;
+import com.liferay.batch.engine.model.BatchEngineTask;
+import com.liferay.batch.engine.service.BatchEngineTaskLocalService;
+import com.liferay.petra.executor.PortalExecutorManager;
+import com.liferay.portal.kernel.cluster.ClusterMasterExecutor;
+import com.liferay.portal.kernel.messaging.BaseMessageListener;
+import com.liferay.portal.kernel.messaging.DestinationNames;
+import com.liferay.portal.kernel.messaging.Message;
+import com.liferay.portal.kernel.messaging.MessageListener;
+import com.liferay.portal.kernel.module.framework.ModuleServiceLifecycle;
+import com.liferay.portal.kernel.scheduler.SchedulerEngineHelper;
+import com.liferay.portal.kernel.scheduler.SchedulerEntry;
+import com.liferay.portal.kernel.scheduler.SchedulerEntryImpl;
+import com.liferay.portal.kernel.scheduler.TimeUnit;
+import com.liferay.portal.kernel.scheduler.Trigger;
+import com.liferay.portal.kernel.scheduler.TriggerFactory;
+import com.liferay.portal.kernel.util.GetterUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Ivica Cardic
+ */
+@Component(
+	immediate = true, property = {"interval=15", "parallel.executions=4"},
+	service = MessageListener.class
+)
+public class BatchEngineTaskListener extends BaseMessageListener {
+
+	@Activate
+	protected void activate(Map<String, Object> properties) {
+		if (_clusterMasterExecutor.isEnabled() &&
+			!_clusterMasterExecutor.isMaster()) {
+
+			return;
+		}
+
+		_checkUnfinishedBatchEngineTasks();
+
+		_activateSchedulerEngine(
+			GetterUtil.getInteger(properties.get("interval")));
+
+		_parallelExecutions = GetterUtil.getInteger(
+			properties.get("parallel.executions"));
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		_schedulerEngineHelper.unregister(this);
+	}
+
+	@Override
+	protected void doReceive(Message message) throws Exception {
+		_startBatchEngineTasks(_parallelExecutions);
+	}
+
+	private void _activateSchedulerEngine(int interval) {
+		Class<?> clazz = getClass();
+
+		String className = clazz.getName();
+
+		Trigger trigger = _triggerFactory.createTrigger(
+			className, className, null, null, interval, TimeUnit.SECOND);
+
+		SchedulerEntry schedulerEntry = new SchedulerEntryImpl(
+			className, trigger);
+
+		_schedulerEngineHelper.register(
+			this, schedulerEntry, DestinationNames.SCHEDULER_DISPATCH);
+	}
+
+	private void _checkUnfinishedBatchEngineTasks() {
+		List<BatchEngineTask> batchEngineTasks =
```

Comment:
@ivicac this is wrong. Component can be dynamic, it can be brought up/down at runtime, when this component activates, there may already have been synced batch tasks running. The logic here would interrupt the current ongoing synced batch tasks.